### PR TITLE
using workspace dependecies

### PR DESCRIPTION
**Pull request by asiniscalchi**
## Summary of Pull Request

- **Moved Evolution and Ownership Nodes**: Relocated evolution and ownership nodes into the root project.
- **CI Pipeline**: Introduced a CI pipeline with GitHub Actions.
- **Cargo Formatting**: Added cargo formatting using the Rust nightly toolchain, following `polkadot-sdk` pattern.
- **Project Reorganization**: Reorganized the project structure and updated Cargo dependencies.
- **Dockerization**: Completed the setup for Docker, and moved all Dockerfiles to the `docker/` directory.
- **Resolved Warnings**: Fixed all cargo warnings and compilation issues.
- **Naming Conventions**: Changed naming from `chain` to `node` in Dockerfile names for clarity.
- **Release Trigger**: Set up GitHub Actions to trigger a release workflow on any tag.

### CODE REVIEW

File: bridge/relays/laos-relay/Cargo.toml [hunk 1]
```diff
@@ -26,31 +26,28 @@ relay-laos-ownership-client = { path="../client-laos-ownership" }
...
 # Substrate Dependencies
-sp-core = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v1.0.0" }
-pallet-sudo = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v1.0.0" }
+sp-core = { workspace = true }
+pallet-sudo = { workspace = true }
 
 [dev-dependencies]
-bp-test-utils = { git = "https://github.com/freeverseio/parity-bridges-common.git", branch = "polkadot-v1.0.0" }
+bp-test-utils = { workspace = true }
 hex-literal = "0.4"
```
Comment (dastansam) [inline bridge/relays/laos-relay/Cargo.toml:52]: not using workspace dependency

File: ownership-chain/pallets/living-assets-ownership/Cargo.toml
Comment (dastanfv) [inline ownership-chain/pallets/living-assets-ownership/Cargo.toml:1]: this pallet has wrong repository link

File: ownership-chain/node/Cargo.toml
Comment (dastanfv) [inline ownership-chain/node/Cargo.toml:1]: same here. wrong repository link and would be nice to use `workspace` level metadata

File: evolution-chain/runtime/Cargo.toml [hunk 2]
```diff
@@ -16,52 +16,51 @@ targets = ["x86_64-unknown-linux-gnu"]
 codec = { package = "parity-scale-codec", version = "3.2.2", default-features = false, features = ["derive"] }
```
Comment (dastanfv) [inline evolution-chain/runtime/Cargo.toml:16]: not using workspace dependency

File: bridge/relays/client-laos-ownership/Cargo.toml [hunk 3]
```diff
@@ -8,15 +8,14 @@ codec = { package = "parity-scale-codec", version = "3.1.5" }
...
+bp-bridge-hub-cumulus = { workspace = true }
+bp-header-chain = { workspace = true }
+bp-messages = { workspace = true }
+bp-polkadot-core = { workspace = true, default-features = false }
+relay-substrate-client = { workspace = true }
+sp-core = { workspace = true }
+sp-runtime = { workspace = true }
+sp-weights = { workspace = true }
 
 bp-laos-evolution = { path = "../../primitives/chain-laos-evolution" }
```
Comment (dastanfv) [inline bridge/relays/client-laos-ownership/Cargo.toml:20]:
```suggestion
bp-laos-evolution = { workspace = true }
```

File: bridge/primitives/chain-laos-ownership/Cargo.toml
Comment (dastanfv) [inline bridge/primitives/chain-laos-ownership/Cargo.toml:1]: `authors`, `edition`, `license` can be borrowed from worksapce

**Review by dastanfv — CHANGES_REQUESTED** [review]
good work in general.

there are two issues:

1. several crates are still using their own defined dependencies, instead of borrowing it from workspace
2. workspace metadata is defined, but not used in the project level Cargo.toml files

File: Cargo.toml [hunk 4]
```diff
@@ -36,6 +39,21 @@ proc-macro2 =  "1.0"
...
+async-trait = "0.1"
+num-format = "0.4"
+num-traits = "0.2"
+rbtag = "0.3"
+rustc-hex = "2.1"
+structopt = "0.3"
+signal-hook = "0.3.16"
+signal-hook-async-std = "0.2.2"
+strum = { version = "0.25.0", features = ["derive"] }
+finality-grandpa = { version = "0.16.2" }
```
Comment (dastanfv) [inline Cargo.toml:56]:
```suggestion
finality-grandpa = "0.16.2"
```

File: Cargo.toml [hunk 5]
```diff
@@ -36,6 +39,21 @@ proc-macro2 =  "1.0"
 quote = "1.0"
 syn = { version = "1.0" }
```
Comment (dastanfv) [inline Cargo.toml:40]:
```suggestion
syn = "1.0"
```

File: bridge/relays/client-rococo/Cargo.toml [hunk 6]
```diff
@@ -6,22 +6,22 @@ edition = "2021"
 license = "GPL-3.0-or-later WITH Classpath-exception-2.0"
 
 [dependencies]
-codec = { package = "parity-scale-codec", version = "3.1.5", features = ["derive"] }
-scale-info = { version = "2.9.0", default-features = false, features = ["derive"] }
+codec = { workspace = true, package = "parity-scale-codec", features = ["derive"] }
+scale-info = { workspace = true, default-features = false, features = ["derive"] }
 subxt = { version = "0.31.0", default-features = false, features = ["native"] }
```
Comment (dastanfv) [inline bridge/relays/client-rococo/Cargo.toml:11]:
```suggestion
subxt = { workspace = true }
```

File: evolution-chain/node/Cargo.toml [hunk 7]
```diff
@@ -17,61 +17,60 @@ targets = ["x86_64-unknown-linux-gnu"]
...
+sp-runtime = { workspace = true }
+sp-io = { workspace = true }
+sp-timestamp = { workspace = true }
+sp-inherents = { workspace = true }
+sp-keyring = { workspace = true }
+frame-system = { workspace = true, features = ["std"] }
+pallet-transaction-payment = { workspace = true }
 
 # These dependencies are used for the node template's RPCs
 jsonrpsee = { version = "0.16.2", features = ["server"] }
```
Comment (dastanfv) [inline evolution-chain/node/Cargo.toml:50]:
```suggestion
jsonrpsee = { workspace = true }
```

File: evolution-chain/runtime/Cargo.toml [hunk 8]
```diff
@@ -98,8 +97,6 @@ std = [
 	"sp-transaction-pool/std",
 	"sp-version/std",
 	"substrate-wasm-builder",
-	"fp-account/std",
-	"pallet-bridge-grandpa/std",
```
Comment (dastanfv) [inline evolution-chain/runtime/Cargo.toml:102]: why are these removed?

File: bridge/relays/laos-relay/Cargo.toml [hunk 9]
```diff
@@ -5,52 +5,49 @@ edition = "2021"
...
-rbtag = "0.3"
-rustc-hex = "2.1"
-structopt = "0.3"
-signal-hook = "0.3.16"
-signal-hook-async-std = "0.2.2"
-strum = { version = "0.25.0", features = ["derive"] }
+anyhow = { workspace = true }
+async-std =  { workspace = true }
+async-trait = { workspace = true }
+codec = { workspace = true, package = "parity-scale-codec", version = "3.1.5" }
```
Comment (dastanfv) [inline bridge/relays/laos-relay/Cargo.toml:11]:
```suggestion
codec = { workspace = true, package = "parity-scale-codec" }
```

**Review by dastanfv — COMMENTED** [review]
looks better but some deps are still not moved to workspace

File: evolution-chain/runtime/Cargo.toml (hunk 8 above)
Comment (asiniscalchi) [reply evolution-chain/runtime/Cargo.toml:102]: they are not a dependency anymore

Comment (dastanfv) [reply evolution-chain/runtime/Cargo.toml:102]: `pallet-bridge-grandpa` is there at line 57

**Comment by asiniscalchi** [comment]
```
warning: /home/asini/workspace/laos/evolution-chain/runtime/Cargo.toml: unused manifest key: dependencies.codec.package
warning: /home/asini/workspace/laos/evolution-chain/pallets/living-assets-evolution/Cargo.toml: unused manifest key: dependencies.codec.package
warning: /home/asini/workspace/laos/bridge/relays/client-laos-evolution/Cargo.toml: unused manifest key: dependencies.codec.package
warning: /home/asini/workspace/laos/bridge/relays/client-laos-ownership/Cargo.toml: unused manifest key: dependencies.codec.package
warning: /home/asini/workspace/laos/bridge/relays/laos-relay/Cargo.toml: unused manifest key: dependencies.codec.package
warning: /home/asini/workspace/laos/bridge/relays/client-rococo/Cargo.toml: unused manifest key: dependencies.codec.package
```
those warnings point to the fact that the workspace dependency is not used. However , using the dependency in the workspace create some compilation error. I let them with the package dependecy.

**Comment by dastanfv** [comment]
> ```
> warning: /home/asini/workspace/laos/evolution-chain/runtime/Cargo.toml: unused manifest key: dependencies.codec.package
> warning: /home/asini/workspace/laos/evolution-chain/pallets/living-assets-evolution/Cargo.toml: unused manifest key: dependencies.codec.package
> warning: /home/asini/workspace/laos/bridge/relays/client-laos-evolution/Cargo.toml: unused manifest key: dependencies.codec.package
> warning: /home/asini/workspace/laos/bridge/relays/client-laos-ownership/Cargo.toml: unused manifest key: dependencies.codec.package
> warning: /home/asini/workspace/laos/bridge/relays/laos-relay/Cargo.toml: unused manifest key: dependencies.codec.package
> warning: /home/asini/workspace/laos/bridge/relays/client-rococo/Cargo.toml: unused manifest key: dependencies.codec.package
> ```
> 
> those warnings point to the fact that the workspace dependency is not used. However , using the dependency in the workspace create some compilation error. I let them with the package dependecy.

there is a work around for this as well. 
1. use `parity-scale-codec` directly everywhere, this means changing code as well.
2. or try this thing that's suggested [here](https://stackoverflow.com/a/76792426)

**Comment by asiniscalchi** [comment]
This PR is intended to make the monorepo works not to fix all the problem that the original repos have.
It's already too much changes.

File: evolution-chain/runtime/Cargo.toml (hunk 8 above)
Comment (asiniscalchi) [reply evolution-chain/runtime/Cargo.toml:102]: pallet-bridge-grandpa was erroneous removed

**Comment by dastanfv** [comment]
> This PR is intended to make the monorepo works not to fix all the problem that the original repos have. It's already too much changes. Feel free to open a PR to fix it when this is merged.

the title of the PR says `using workspace dependencies` and I am leaving comments based on that. And those warnings are introduced after the workspace dependency switch, so it is relevant for this PR. I am ok with leaving the warnings of course, but an issue needs to be created

**Comment by dastanfv** [comment]
> > This PR is intended to make the monorepo works not to fix all the problem that the original repos have. It's already too much changes. Feel free to open a PR to fix it when this is merged.
> 
> the title of the PR says `using workspace dependencies` and I am leaving comments based on that. And those warnings are introduced after the workspace dependency switch, so it is relevant for this PR. I am ok with leaving the warnings of course, but an issue needs to be created

https://github.com/freeverseio/laos/issues/17

**Review by dastanfv — APPROVED** [review]
